### PR TITLE
Faster jupyter notebook image

### PR DESCRIPTION
**Pull request by paulgb**
Instead of doing the config at startup time, we do it as part of the build step. `run.sh` becomes `config.sh`.

Also, now that we have bearer tokens, the ability to set a bearer token has been replaced with a link to the bearer token docs.

### CODE REVIEW

**Review by pretentious7 — APPROVED** [review]
everything looks good, guess we can merge this as soon as we get the perf regressions figured out.

File: README.md [hunk 1]
```diff
@@ -5,16 +5,16 @@ Base repo for running Jupyter Notebooks on the Jamsocket platform
 
 1. Fork this repository.
 2. Add your notebook(s) to the `/notebook` directory.
-3. Add your notebook's python dependencies to `requirements.txt`.
-4. Update the `NOTEBOOK` value in `run.sh` to make sure Jupyter opens your notebook on start up.
+3. Add your notebook's python dependencies to `requirements.txt` (alternatively, if you prefer `pipenv`, add them with `pipenv install` and then run `pipenv run freeze`).
+4. Update the `NOTEBOOK` value in `config.sh` to make sure Jupyter opens your notebook on start up.
```
Comment (rolyatmax) [inline README.md]: Is `config.sh` supposed to be part of these changes? I don't think it got checked in?

**Comment by rolyatmax** [comment]
Is the goal with this PR to make the Jupyter notebook start faster? If so, does it do that simply by using a more recent version of Jupyter notebooks (which happens to not support the default notebook functionality which we've now added upstream?)?

**Review by pretentious7 — APPROVED** [review]
Looks good to me.

So the lock.sh thing is only needed for updating the notebook right?

File: README.md (hunk 1 above)
Comment (paulgb) [reply README.md]: it's removed, will update readme

**Comment by paulgb** [comment]
> Is the goal with this PR to make the Jupyter notebook start faster? If so, does it do that simply by using a more recent version of Jupyter notebooks (which happens to not support the default notebook functionality which we've now added upstream?)?

it uses jupyter notebooks v7 instead of the deprecated v6, so it's a better foundation to start optimizing from. We avoid doing some things at runtime to create config that we were doing before, but it feels about the same to me so far.

**Comment by paulgb** [comment]
> Looks good to me.
> 
> So the lock.sh thing is only needed for updating the notebook right?

yes, it creates `freeze.txt`. I've renamed it to `freeze.sh`

**Pull request merged**